elasticsearch/src/main/java/com/gentics/mesh/search/index/node/NodeContainerTransformer.java

```diff
@@ -131,15 +130,20 @@ private void addParentNodeInfo(JsonObject document, HibNode parentNode) {
 	private void addPermissionInfo(JsonObject document, HibNode node, ContainerType type) {
 		List<String> roleUuids = new ArrayList<>();
 
-		for (HibRole role : toGraph(node).getRolesWithPerm(READ_PERM)) {
-			roleUuids.add(role.getUuid());
-		}
-
-		// Also add the roles which would grant read on published nodes if the container is published.
-		if (type == PUBLISHED) {
-			for (HibRole role : toGraph(node).getRolesWithPerm(READ_PUBLISHED_PERM)) {
+		try {
+			for (HibRole role : Tx.get().roleDao().getRolesWithPerm(node, READ_PERM)) {
 				roleUuids.add(role.getUuid());
 			}
+
+			// Also add the roles which would grant read on published nodes if the container is published.
+			if (type == PUBLISHED) {
+				for (HibRole role : Tx.get().roleDao().getRolesWithPerm(node, READ_PUBLISHED_PERM)) {
+					roleUuids.add(role.getUuid());
+				}
+			}
+		} catch (NullPointerException e) {
```

Comment:
A questionnable part. During the docu generation either `Tx.get()` or `roleDao()` may return null.